src/data/repositories/UserD2ApiRepository.ts

```diff
@@ -239,10 +247,11 @@ export class UserD2ApiRepository implements UserRepository {
 
     private toDomainUser(input: ApiUser): User {
         const { userCredentials, ...user } = input;
-        const authorities = _(userCredentials.userRoles.map(userRole => userRole.authorities))
-            .flatten()
-            .uniq()
-            .value();
+        const authorities =
+            _(userCredentials.userRoles?.map(userRole => userRole.authorities))
```

Comment:
[minor] Lodash accepts null/undefined values, it considers them empty arrays. So this can be written: `_(userCredentials.userRoles).map(userRole => userRole.authorities)`